src/library.js

```diff
@@ -2386,6 +2386,37 @@ addToLibrary({
   _emscripten_get_now_is_monotonic__deps: ['$nowIsMonotonic'],
   _emscripten_get_now_is_monotonic: () => nowIsMonotonic,
 
+  __emscripten_atomics_sleep__internal: true,
+#if SHARED_MEMORY
+  // When SHARED_MEMORY is true, this isn't used, but we need it here to make
+  // gen_sig_info happy.
+  __emscripten_atomics_sleep: (ms) => {},
+#else
+  // In a browser without cross origin isolation, SharedArrayBuffer is deleted
+  // from the global scope:
+  // https://html.spec.whatwg.org/multipage/webappapis.html#creating-a-new-javascript-realm
+  // However, it says:
+  //
+  // > Web developers can still get at the constructor through
+  // > `new WebAssembly.Memory({ shared:true, initial:0, maximum:0}).buffer.constructor`.
+  //
+  // That is what we are up to here. Use an IIFE to avoid shadowing
+  // SharedArrayBuffer globally.
+  __emscripten_atomics_sleep__postset: `
+    var SABConstructor = new WebAssembly.Memory({"shared":true,"initial":0,"maximum":0}).buffer.constructor;
+    var waitBuffer = new Int32Array(new SABConstructor(4));
+    var _supports_atomics_wait;
+    try {
+      ___emscripten_atomics_sleep(0);
+      _supports_atomics_wait = true;
+    } catch (e) {
+      _supports_atomics_wait = false;
```

Comment:
This feature test code gets unconditionally emitted to all generated code that calls to sleep.

There is a feature_matrix.py that already has a feature check for when users are targeting any environments that potentially have SAB:

```
  Feature.THREADS: {
    'chrome': 74,
    'firefox': 79,
    'safari': 140100,
  },
```
I think if those match, we could avoid doing a feature test here - or if targeting an old enough combination, we could statically know that support won't be there.

Also instead of 
```js
   try {
       ___emscripten_atomics_sleep(0);
       _supports_atomics_wait = true;
     } catch (e) {
       _supports_atomics_wait = false;
    }
```
recommend
```js
   try {
       ___emscripten_atomics_sleep(0);
       _supports_atomics_wait = 1;
     } catch (e) { }
```

Reply:
Since is a performance improvement, not sure why people wouldn't want to use it when possible, even if they also want to support old browsers.

Reply:
To clarify, I was proposing to add the check both ways:
 a) if targeting a new enough browser, then such a feature check could be omitted and the feature always used unconditionally, and
 b) if targeting too old browsers, the feature could be skipped on the basis that it won't be there.
Two reasons that come to mind are code size, and static linters that look for uses of different APIs and reject the code if it refers to any unallowed constructs.

Although now I realize that the feature matrix would not allow doing b) after all, only a).